cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -224,10 +225,57 @@ public SparkClassRewriter(Function<String, InputStream> resourceLookup, boolean
       // Rewrite Spark DiskStore class and classes in the network package for Netty 4.1 compatibility
       return rewriteSparkNetworkClass(input);
     }
+    if (className.equals(SPARK_OUTPUT_METRICS.getClassName())) {
+      // Rewrite the Spark OutputMetrics to skip overwriting bytes written metrics with 0.
+      return rewriteOutputMetrics(SPARK_OUTPUT_METRICS, input);
+    }
 
     return null;
   }
 
+  /**
+   * Rewrites the OutputMetrics#setBytesWritten method to skip recording zero (0) value to avoid overwritting
```

Comment:
nit: overwritting -> overwriting

Also curious, when does something call setBytesWritten(0) after setting it to a non-zero value? Is this just a Spark bug?

Reply:
Spark does it by blinding calling setBytesWritten using FileSystem statistics. It is asymmetric in Spark in how it implements bytes read (which only applies to file inputs) and bytes written (which blindly set).